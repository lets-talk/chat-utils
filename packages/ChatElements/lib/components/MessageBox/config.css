@import '../../common/styles/variables.css';

:root {
  --messageBox-background: var(--palette-orange-900);
  --messageBox-color: var(--color-white);
  --messageBox-title-color: var(--color-white);
  --messageBox-time-color: var(--color-white);
  --messageBox-side-margin: 10px;
  --messageBox-border-radius: 5px;
  --messageBox-box-shadow: 0px rgba(0, 0, 0, .2);
}
