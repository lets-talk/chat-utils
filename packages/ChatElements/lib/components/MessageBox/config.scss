@import "../../common/styles/variables";
@import "../../common/styles/themes/themify";

$messageBox-font-size: $font-size-small;
$messageBox-title-font-size: $font-size-normal;
$messageBox-title-color: $color-white;
$messageBox-time-font-size: $font-size-tiny;

$messageBox-side-margin: 10px;
$messageBox-border-radius: 5px;
$messageBox-box-shadow: none;
