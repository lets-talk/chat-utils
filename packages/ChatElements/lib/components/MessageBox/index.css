@import './config.css';

.letstalk-container-mbox {
    flex-direction: column;
    display: block;
    overflow: hidden;
    min-width: 200px;
}

.letstalk-container-mbox .letstalk-mbox-avatar-container{
  display: inline-block;
}

.letstalk-mbox-avatar-container-left {
  float: left;
}

.letstalk-mbox-avatar-container-right {
  float: right;
}

.letstalk-mbox-forward {
    width: 30px;
    height: 30px;
    border-radius: 20px;
    background: #fff;
    position: absolute;
    flex-direction: row;
    align-self: center;
    align-items: center;
    justify-content: center;
    box-shadow: 0 0 5px 0 rgba(164, 164, 164, 1);
    cursor: pointer;
    transition: all 0.3s ease;
    top: 0;
    bottom: 0;
    margin: auto;
}

.letstalk-mbox-forward-left {
    display: flex;
    opacity: 0;
    visibility: hidden;
    left: -50px;
}

.letstalk-mbox-forward-right {
    display: flex;
    opacity: 0;
    visibility: hidden;
    right: -50px;
}

.letstalk-container-mbox:hover .letstalk-mbox-forward-left {
    opacity: 1;
    visibility: visible;
}

.letstalk-container-mbox:hover .letstalk-mbox-forward-right {
    opacity: 1;
    visibility: visible;
}

.letstalk-mbox {
    position: relative;
    background-color: var(--messageBox-background);
    color: var(--messageBox-color);
    border-radius: var(--messageBox-border-radius);
    box-shadow: var(--messageBox-box-shadow);
    margin-left: var(--messageBox-side-margin);
    margin-right: 5px;
    margin-top: 3px;
    flex-direction: column;
    margin-bottom: 3px;
    padding: 6px 9px 8px 9px;
    float: left;
    min-width: 140px;
}

.letstalk-mbox-body {
    margin: 0;
    padding: 0;
    position: relative;
}

.letstalk-mbox.letstalk-mbox-right {
    float: right;
    margin-left: 5px;
    margin-right: var(--messageBox-side-margin);
}

.letstalk-mbox-text {
    font-size: 13.6px;
    word-break: break-word;
}

.letstalk-mbox-text:after {
    content: "\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0\00a0"
}

.letstalk-mbox-time {
    text-align: right;
    color: var(--messageBox-time-color);
    font-size: var(--font-size-tiny);
    position: relative;
    right: -4px;
    bottom: -5px;
}

.letstalk-mbox-time-block {
    /*position: relative;*/
    right: 0;
    bottom: 0;
    left: 0;
    margin-right: -6px;
    margin-left: -6px;
    padding-top: 5px;
    padding-right: 3px;
    padding-bottom: 2px;
    background: linear-gradient(to top, rgba(0,0,0,0.33), transparent);
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
    color: #fff;
}

.letstalk-mbox--clear-padding {
    padding-bottom: 3px;
}

.letstalk-mbox-title {
    margin: 0;
    margin-bottom: 8px;
    font-weight: 500;
    font-size: 13px;
    color: var(--messageBox-title-color);
    user-select: none;
    cursor: pointer;
    display: flex;
    align-items: center;
}

.letstalk-mbox-title:hover {
    text-decoration: underline;
}

.letstalk-mbox-title--clear {
    margin-bottom: 5px;
}

.letstalk-mbox-status {
    margin-left: 3px;
    font-size: 15px;
}

.letstalk-mbox-title > .letstalk-avatar-container {
    margin-right: 5px;
}
