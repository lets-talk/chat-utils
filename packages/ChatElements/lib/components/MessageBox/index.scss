@import './config';

.letstalk-container-mbox {
    position: relative;
    flex-direction: column;
    display: block;
    overflow: hidden;
    min-width: 200px;
}

.letstalk-container-mbox {
  .letstalk-mbox-avatar-container{
    display: inline-block;

    &.letstalk-mbox-avatar-container-left {
      position: absolute;
      bottom: 0;
      left: 0;
    }

    &.letstalk-mbox-avatar-container-right {
      position: absolute;
      bottom: 0;
      right: 0;
    }
  }
}

.letstalk-mbox-avatar-container-left {
  float: left;
}

.letstalk-mbox-avatar-container-right {
  float: right;
}

.letstalk-mbox-forward {
    width: 30px;
    height: 30px;
    border-radius: 20px;
    background: #fff;
    position: absolute;
    flex-direction: row;
    align-self: center;
    align-items: center;
    justify-content: center;
    box-shadow: 0 0 5px 0 rgba(164, 164, 164, 1);
    cursor: pointer;
    transition: all 0.3s ease;
    top: 0;
    bottom: 0;
    margin: auto;
}

.letstalk-mbox-forward-left {
    display: flex;
    opacity: 0;
    visibility: hidden;
    left: -50px;
}

.letstalk-mbox-forward-right {
    display: flex;
    opacity: 0;
    visibility: hidden;
    right: -50px;
}

.letstalk-container-mbox:hover .letstalk-mbox-forward-left {
    opacity: 1;
    visibility: visible;
}

.letstalk-container-mbox:hover .letstalk-mbox-forward-right {
    opacity: 1;
    visibility: visible;
}

.letstalk-mbox {
  position: relative;
  @include themify($themes) {
    color: themed('color-text');
    background-color: themed('color-background-primary');
  }
  border-radius: $messageBox-border-radius;
  box-shadow: $messageBox-box-shadow;
  margin-left: $messageBox-side-margin;
  margin-right: 5px;
  margin-top: 3px;
  flex-direction: column;
  margin-bottom: 3px;
  padding: 15px;

  &.letstalk-mbox-left {
    float: left;
    margin-left: 5px;
    margin-left: 45px;
  }

  &.letstalk-mbox-right {
    float: right;
    margin-left: 5px;
    margin-right: 45px;
  }

  .letstalk-mbox-time {
    display: inline-block;
    width: 50px;
    text-align: right;
    font-size: $font-size-caption;
    position: relative;

    @include themify($themes) {
      color: themed('color-hint-secondary');
    }

    &.letstalk-mbox-time-block {
      float: right;
    }
  }

  &.letstalk-mbox-agent,
  &.letstalk-mbox-bot {
    @include themify($themes) {
      color: themed('color-primary');
      background-color: themed('color-background-secondary');
    }
    .letstalk-mbox-title {
      @include themify($themes) {
        color: themed('color-bot-agent');
      }
    }
    .letstalk-mbox-time {
      @include themify($themes) {
        color: themed('color-hint-primary');
      }
    }
  }
}

.letstalk-mbox.letstalk-mbox-actionable {
  padding: 0 2px 0 15px;
  border-radius: 5px;
  box-shadow: 0px 0px 2px 1px rgba(94,124,139,0.3);
  background-color: $color-white;
  min-width: 50%;
  @include themify($themes) {
    color: themed('color-primary');
  }
}

.letstalk-mbox-body {
  margin: 0;
  padding: 0;
  position: relative;
}

.letstalk-mbox-text,
.letstalk-mbox-typing {
    word-break: break-word;
}

.letstalk-mbox-content:not(.letstalk-mbox-typing) {
  display: inline-block;
  width: calc(100% - 50px);
}

.letstalk-mbox--clear-padding {
  padding-bottom: 3px;
}

.letstalk-mbox-title {
  margin: 0;
  margin-bottom: 8px;
  font-weight: $messageBox-title-font-weight;
  font-size: $messageBox-title-font-size;
  color: $messageBox-title-color;
  user-select: none;
  cursor: pointer;
  display: flex;
  align-items: center;
}

.letstalk-mbox-title:hover {
  text-decoration: underline;
}

.letstalk-mbox-title--clear {
  margin-bottom: 5px;
}

.letstalk-mbox-status {
  margin-left: 3px;
}

.letstalk-mbox-title > .letstalk-avatar-container {
  margin-right: 5px;
}
