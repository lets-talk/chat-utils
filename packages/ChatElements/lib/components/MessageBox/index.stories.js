import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, object, select } from '@storybook/addon-knobs';

import { wrapWithThemes } from '../../utils/stories';
// Component to show on storybook
import MessageBox from '.';


const stories = storiesOf('MessageBox', module);
stories.addDecorator(withKnobs);

const themeOptions = { default: 'DefaultTheme', light: 'LightTheme', dark: 'DarkTheme' };
const defaultTheme = 'default';

// This HOC Wraps the Box so it is inside a ThemeProvider for styled-components
// and also wraps it inside a div with the selected global themes
const MessageBoxWithTheme = wrapWithThemes(MessageBox);

const messages = [
  {
    position: 'left',
    forwarded: true,
    type: 'text',
    theme: 'white',
    view: 'list',
    title: 'TalkBot',
    text: 'Buenos días! En que podemos ayudarte?',
    status: 'waiting',
    date: '2017-11-15T20:34:27.173Z',
    dateString: '18:23',
    user_type: 'bot',
  },
  {
    position: 'right',
    forwarded: true,
    type: 'file',
    theme: 'white',
    view: 'list',
    title: 'Solicitud de nueva tarjeta Banco',
    text: 'Solicitud de nueva tarjeta Banco',
    data: {
      uri: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
      status: {
        click: true,
        loading: 70,
      },
      width: 300,
      height: 300,
      size: '2Mb',
    },
    status: 'sent',
    date: '2017-11-15T20:34:27.173Z',
    dateString: '17:34',
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
  },
  {
    position: 'right',
    forwarded: false,
    type: 'photo',
    theme: 'white',
    view: 'list',
    title: 'Solicitud de nueva tarjeta Banco',
    text: 'Solicitud de nueva tarjeta Banco',
    data: {
      uri: 'http://i46.tinypic.com/sexbb8.pngg',
      status: {
        click: true,
        loading: 70,
      },
      width: 150,
      height: 150,
      size: '2Mb',
    },
    status: 'sent',
    date: '2017-11-15T20:34:27.173Z',
    dateString: '17:34',
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
  },
  {
    position: 'left',
    forwarded: false,
    type: 'actionable',
    theme: 'white',
    view: 'list',
    text: 'Datos Contacto',
    data: {
      actions: [
        {
          id: 1,
          name: 'Emergencia Bancaria',
        },
        {
          id: 2,
          name: 'Tarjeta de crédito',
        },
        {
          id: 3,
          name: 'Crédito Personal',
        },
        {
          id: 4,
          name: 'Otro',
        },
      ],
    },
    status: 'waiting',
    date: '2017-11-15T20:34:27.173Z',
    dateString: '18:23',
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
  },
  {
    position: 'right',
    forwarded: true,
    type: 'typing',
    theme: 'white',
    view: 'list',
    title: '',
    text: '',
    status: 'read',
    date: '2017-11-14T18:25:22.133Z',
    dateString: '18:25',
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    user_type: 'agent',
  },
];

stories.addWithInfo(
  'MessageBox - Text Message',
  'TextMessage type use',
  () => (
    <div style={{ position: 'fixed', bottom: 0, right: '10px' }}>
      <MessageBoxWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
        {...messages[0]}
        onOpen={action('onOpen')}
        onDownload={action('onDownload')}
        onTitleClick={action('onTitleClick')}
        onForwardClick={action('onForwardClick')}
        onClick={action('onClick')}
        onActionMessageClick={action('onActionMessageClick')}
      />
    </div>
  )
);

stories.addWithInfo(
  'MessageBox - Fie Message Type',
  'FileMessage type use',
  () => (
    <div style={{ position: 'fixed', bottom: 0, right: '10px' }}>
      <MessageBoxWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
        {...messages[1]}
        onOpen={action('onOpen')}
        onDownload={action('onDownload')}
        onTitleClick={action('onTitleClick')}
        onForwardClick={action('onForwardClick')}
        onClick={action('onClick')}
        onActionMessageClick={action('onActionMessageClick')}
      />
    </div>
  )
);

stories.addWithInfo(
  'MessageBox - Photo Message Type',
  'Photo type use',
  () => (
    <div style={{ position: 'fixed', bottom: 0, right: '10px' }}>
      <MessageBoxWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
        {...messages[2]}
        onOpen={action('onOpen')}
        onDownload={action('onDownload')}
        onTitleClick={action('onTitleClick')}
        onForwardClick={action('onForwardClick')}
        onClick={action('onClick')}
        onActionMessageClick={action('onActionMessageClick')}
      />
    </div>
  )
);

stories.addWithInfo(
  'MessageBox - Actionable Message Type',
  'Message of ActionableMessageBox type',
  () => (
    <div style={{ position: 'fixed', bottom: 0, right: '10px' }}>
      <MessageBoxWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
        {...messages[3]}
        onOpen={action('onOpen')}
        onDownload={action('onDownload')}
        onTitleClick={action('onTitleClick')}
        onForwardClick={action('onForwardClick')}
        onClick={action('onClick')}
        onActionMessageClick={action('onActionMessageClick')}
      />
    </div>
  )
);

stories.addWithInfo(
  'MessageBox - Typing Message Type',
  'Message of Typing type',
  () => (
    <div style={{ position: 'fixed', bottom: 0, right: '10px' }}>
      <MessageBoxWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
        {...messages[4]}
        onOpen={action('onOpen')}
        onDownload={action('onDownload')}
        onTitleClick={action('onTitleClick')}
        onForwardClick={action('onForwardClick')}
        onClick={action('onClick')}
        onActionMessageClick={action('onActionMessageClick')}
      />
    </div>
  )
);
