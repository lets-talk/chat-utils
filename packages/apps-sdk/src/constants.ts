// Apps SDK Events
export const EVENT_TYPE_LOAD_APP = 'launcher-load-app';
export const EVENT_TYPE_REMOVE_APP = 'launcher-remove-app';
export const EVENT_TYPE_GET_APP_SETTINGS = 'launcher-get-app-settings';
export const EVENT_TYPE_NOTIFY_APP_EVENT = 'launcher-notify-app-event';