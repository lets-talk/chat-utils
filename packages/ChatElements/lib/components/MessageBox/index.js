import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import moment from 'moment';

// Icons to be used on this component
import FaForward from 'react-icons/lib/fa/mail-forward';
import IoDoneAll from 'react-icons/lib/io/android-done-all';
import MdIosTime from 'react-icons/lib/md/access-time';
import MdCheck from 'react-icons/lib/md/check';

// Constants
import constants from '../../utils/constants';

// Other components used by this component
import FileMessage from '../FileMessage';
import SystemMessage from '../SystemMessage';
import TimeMarkMessage from '../TimeMarkMessage';
import PhotoMessage from '../PhotoMessage';
import Avatar from '../Avatar';
import Loader from '../Loader';
import ActionableMessageBox from '../ActionableMessageBox';

// Component styles
import './index.scss';

const MessageBox = (props) => {
  const { messagesTypes, personTypes } = constants;
  const mboxExtraClasses = classNames('letstalk-mbox', `letstalk-mbox-${props.position}`, {
    'letstalk-mbox-agent': props.person.type === personTypes.AGENT,
    'letstalk-mbox-bot': props.person.type === personTypes.BOT && props.type !== messagesTypes.ACTIONABLE,
    'letstalk-mbox-actionable': props.type === messagesTypes.ACTIONABLE,
  });

  const thatAbsoluteTime = props.type !== messagesTypes.TEXT && props.type !== messagesTypes.TYPING;
  const automaticMessage = () => {
    if (props.type === messagesTypes.SYSTEM) {
      return (<SystemMessage
        text={props.text}
      />);
    } else if (props.type === messagesTypes.TIME) {
      return (<TimeMarkMessage
        text={props.text}
      />);
    }
    return <div />;
  };

  return (
    <div
      role="button"
      tabIndex="-3"
      className={classNames('letstalk-container-mbox', props.className)}
      onClick={props.onClick}
      onKeyPress={props.onClick}
    >
      {
        props.person.avatar && props.type !== messagesTypes.SYSTEM &&
        <div
          className={classNames(
            'letstalk-mbox-avatar-container',
            { 'letstalk-mbox-avatar-container-left': props.position === 'left' },
            { 'letstalk-mbox-avatar-container-right': props.position === 'right' }
          )}
        >
          <Avatar
            src={props.person.avatar}
            withStatus={false}
            size="xsmall"
            status="online"
          />
        </div>
      }
      {
        props.type === messagesTypes.SYSTEM || props.type === messagesTypes.TIME ?
          automaticMessage()
          :
          <div
            className={classNames(
              mboxExtraClasses,
              { 'letstalk-mbox--clear-padding': thatAbsoluteTime },
            )}
          >
            <div className="letstalk-mbox-body">
              {
                props.forwarded === true &&
                  <div
                    role="button"
                    tabIndex="-2"
                    className={classNames(
                      'letstalk-mbox-forward',
                      { 'letstalk-mbox-forward-right': props.position === 'left' },
                      { 'letstalk-mbox-forward-left': props.position === 'right' }
                    )}
                    onClick={props.onForwardClick}
                    onKeyPress={props.onForwardClick}
                  >
                    <FaForward />
                  </div>
              }

              {
                (props.title) &&
                  <div
                    role="button"
                    tabIndex="-1"
                    onClick={props.onTitleClick}
                    onKeyPress={props.onTitleClick}
                    className={classNames('letstalk-mbox-title', {
                      'letstalk-mbox-title--clear': props.type === messagesTypes.TEXT,
                    })}
                  >
                    {
                      props.title &&
                      <span>{props.title}</span>
                    }
                  </div>
              }

              {
                props.type === messagesTypes.TEXT &&
                  <div className={classNames('letstalk-mbox-content', 'letstalk-mbox-text')}>
                    {props.text}
                  </div>
              }

              {
                props.type === messagesTypes.TYPING &&
                  <div className={classNames('letstalk-mbox-content', 'letstalk-mbox-typing')}>
                    <Loader active type="ball-pulse" size="xsmall" color="rgb(113, 131, 150)" />
                  </div>
              }

              {
                props.type === messagesTypes.PHOTO &&
                <PhotoMessage
                  onOpen={props.onOpen}
                  onDownload={props.onDownload}
                  data={props.data}
                  text={props.text}
                />
              }

              {
                props.type === messagesTypes.FILE &&
                  <FileMessage
                    onOpen={props.onOpen}
                    onDownload={props.onDownload}
                    data={props.data}
                    text={props.text}
                  />
              }

              {
                props.type === messagesTypes.ACTIONABLE &&
                  <ActionableMessageBox
                    data={props.data}
                    onClickAction={props.onActionMessageClick}
                  />
              }

              {props.type !== messagesTypes.TYPING && props.type !== messagesTypes.ACTIONABLE &&
                <div className={classNames('letstalk-mbox-time', { 'letstalk-mbox-time-block': thatAbsoluteTime })}>
                  {
                    props.date && (props.type !== messagesTypes.TYPING) &&
                    (
                      props.dateString ||
                        moment(props.date).fromNow()
                    )
                  }
                  {
                    props.status &&
                      <span className="letstalk-mbox-status">
                        {
                          props.status === 'waiting' &&
                          <MdIosTime />
                        }

                        {
                          props.status === 'sent' &&
                          <MdCheck />
                        }

                        {
                          props.status === 'received' &&
                          <IoDoneAll />
                        }

                        {
                          props.status === 'read' &&
                          <IoDoneAll color="#4FC3F7" />
                        }
                      </span>
                  }
                </div>
              }

            </div>
          </div>
      }
    </div>
  );
};

MessageBox.propTypes = {
  /**
   * Position: What side the message is displayed inside the chatbox
   */
  position: PropTypes.string,
  /**
   * Type of Message: Type of the message. Values are defined as constants.
   * The current supported types are: SYSTEM, TEXT, TYPING, TIME, ACTIONABLE, FILE, PHOTO.
   */
  type: PropTypes.string,
  /**
   * Text: Actual message content text.
   */
  text: PropTypes.string,
  /**
   * Title: Message title. Is used to display as a first line. It is also possible to make it clickable
   */
  title: PropTypes.string,
  /**
   * onTitleClick: Handler function to be called when user click on message title
   */
  onTitleClick: PropTypes.func,
  /**
   * onForwardClick: Handler function to be called when user click on forwared message
   */
  onForwardClick: PropTypes.func,
  /**
   * date: Message creation date.
   */
  date: PropTypes.instanceOf(Date),
  /**
   * data: Object with extra data used to display information about the message
   */
  data: PropTypes.object,
  /**
   * onClick: Handler function that is called on message click action
   */
  onClick: PropTypes.func,
  /**
   * onActionMessageClick: Handler function that is called when user click on an item of
   * a message that contains actionable action list
   */
  onActionMessageClick: PropTypes.func,
  /**
   * onOpen: Handler function that is called when user tries to open a File/Photo message type
   */
  onOpen: PropTypes.func,
  /**
  * onDownload: Handler function that is called when user tries to download a File/Photo message type
  */
  onDownload: PropTypes.func,
  /**
  * forwarded: Boolean that indicates if message was forwareded. If this is false (default) onForwardClick
  * handler has no effect at all
  */
  forwarded: PropTypes.bool,
  /**
   * status: Message status. Values are defined as constants.
   * Current supported values are: WAITING, SENT, RECEIVED, READ
   */
  status: PropTypes.string,
  /**
   * dateString: Formated date string to show. This is the default to use.
   * If this is not provided by default to show the time moment(props.date).fromNow() is used.
   */
  dateString: PropTypes.string,
  /**
   * person: Object representing the person that created or submited this message.
   */
  person: PropTypes.object,
  /**
   * className: Extra className to provide to the component in order to style it when used in different contexts.
   */
  className: PropTypes.string,
};

MessageBox.defaultProps = {
  position: 'left',
  type: 'text',
  text: '',
  title: null,
  onTitleClick: null,
  onActionMessageClick: null,
  onForwardClick: null,
  date: new Date(),
  data: {},
  onClick: null,
  onOpen: null,
  onDownload: null,
  forwarded: false,
  status: '',
  dateString: null,
  person: null,
  className: '',
};


export default MessageBox;
