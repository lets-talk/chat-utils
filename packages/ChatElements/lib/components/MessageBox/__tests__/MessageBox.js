import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MessageBox from '../index';
// Constants
import constants from '../../../utils/constants';

describe('MessageBox component', () => {
  const person = {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Client',
  };

  const { messagesTypes } = constants;

  it('should render without issues', () => {
    const component = shallow(<MessageBox status="read" person={person} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render a System message', () => {
    const props = {
      type: messagesTypes.SYSTEM,
      status: 'read',
      person,
    };
    const component = shallow(<MessageBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render without issues', () => {
    const props = {
      type: messagesTypes.TIME,
      status: 'read',
      person,
    };

    const component = shallow(<MessageBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
