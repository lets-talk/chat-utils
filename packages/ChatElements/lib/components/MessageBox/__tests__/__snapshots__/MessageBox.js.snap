// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox component should render without issues 1`] = `
<div
  className="letstalk-container-mbox"
  onClick={null}
  onKeyPress={null}
  role="button"
  tabIndex="-3"
>
  <div
    className="letstalk-mbox letstalk-mbox-left"
  >
    <div
      className="letstalk-mbox-body"
    >
      <div
        className="letstalk-mbox-content letstalk-mbox-text"
      />
      <div
        className="letstalk-mbox-time"
      >
        a few seconds ago
        <span
          className="letstalk-mbox-status"
        >
          <IoAndroidDoneAll
            color="#4FC3F7"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
